src/test_tube/mod.rs

```diff
@@ -0,0 +1,113 @@
+mod modules;
+mod tests;
+
+use crate::test_tube::modules::{Authz, SingleSidedLpCl};
+
+use cosmwasm_std::{Coin, Uint128};
+use osmosis_std::types::osmosis::concentratedliquidity::v1beta1::{
+    CreateConcentratedLiquidityPoolsProposal, MsgCreatePosition, Pool, PoolRecord, PoolsRequest,
+};
+use prost::Message;
+
+use crate::tick::{MAX_TICK, MIN_INITIALIZED_TICK};
+use osmosis_test_tube::{
+    Account, Bank, ConcentratedLiquidity, GovWithAppAccess, Module, OsmosisTestApp,
+};
+
+pub const UFOO: &'static str =
+    "ibc/0CD3A0285E1341859B5E86B6AB7682F023D03E97607CCC1DC95706411D866DF7";
+pub const UBAR: &'static str =
+    "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2";
```

Comment:
These denoms are used since they are authorized quote assets so I didn't need to fight with adding ufoo and ubar to the params. I did not use uosmo because we pay tx fees with uosmo, so it makes it simpler to calculate how much was actually used in the swap and position creation.